lib/auth/userloginstate/generator.go

```diff
@@ -186,19 +191,31 @@ func (g *Generator) addAccessListsToState(ctx context.Context, user types.User,
 	}
 
 	for _, accessList := range accessLists {
-		if err := services.IsAccessListOwner(identity, accessList); err == nil {
-			g.grantRolesAndTraits(identity, accessList.Spec.OwnerGrants, state)
+		if dynamic, err := services.IsAccessListOwner(ctx, g.accessLists, identity, accessList); err == nil {
+			added := g.grantRolesAndTraits(identity, accessList.Spec.OwnerGrants, state)
+			if dynamic {
```

Comment:
if dynamic is true it doesn't mean that a user is a owner/member vie nested access list The = `services.IsAccessListOwner` will return true for top parent lever owner/membership user  assignments.